mesonbuild/interpreter.py

```diff
@@ -2954,8 +2961,19 @@ def func_project(self, node, args, kwargs):
         self.build.projects[self.subproject] = proj_name
         mlog.log('Project name:', mlog.bold(proj_name))
         mlog.log('Project version:', mlog.bold(self.project_version))
-        self.add_languages(proj_langs, True, MachineChoice.BUILD)
-        self.add_languages(proj_langs, True, MachineChoice.HOST)
+
+        if 'languages' in kwargs:
+            if len(proj_langs) > 0:
+                raise InterpreterException('project() may not have both language positional arguments and languages:.')
+            self.add_languages(mesonlib.stringlistify(kwargs.get('languages', [])), True, MachineChoice.HOST)
+        else:
+            self.add_languages(proj_langs, True, MachineChoice.HOST)
+
+        if 'native_languages' in kwargs:
+            self.add_languages(mesonlib.stringlistify(kwargs.get('native_languages', [])), True, MachineChoice.BUILD)
+        else:
+            self.add_languages(proj_langs, False, MachineChoice.BUILD)
+
```

Comment:
When there are positional languages and `native_languages`, there is no error, but there should be.

How about:
```suggestion

        if len(proj_langs) > 0 and ('native_languages' in kwargs or 'languages' in kwargs):
                raise InterpreterException('project() may not have both language positional arguments and languages:.')
        self.add_languages(proj_langs, True, MachineChoice.BUILD)
        self.add_languages(proj_langs, True, MachineChoice.HOST)
        self.add_languages(mesonlib.stringlistify(kwargs.get('native_languages', [])), True, MachineChoice.BUILD)
        self.add_languages(mesonlib.stringlistify(kwargs.get('languages', [])), True, MachineChoice.HOST)
```

Reply:
Oh nevermind I misread the message